security/libkmip-tests.py

```diff
@@ -31,16 +31,31 @@ def setUp(self):
         '''
         # Check for basic utilities
         smm = SoftwareManager()
+        detected_distro = distro.detect()
         for package in ['gcc', 'make', 'autoconf']:
             if not smm.check_installed(package) and not smm.install(package):
                 self.cancel('%s is needed for the test to be run' % package)
-        url = "https://github.com/OpenKMIP/libkmip/archive/refs/heads/master.zip"
-        tarball = self.fetch_asset(url, expire='7d')
-        archive.extract(tarball, self.workdir)
-        self.sourcedir = os.path.join(self.workdir, 'libkmip-master')
-        if build.make(self.sourcedir):
-            self.fail("Failed to compile libkmip")
-        os.chdir(self.sourcedir)
+        run_type = self.params.get('type', default='upstream')
+        if run_type == "upstream":
+            default_url = ("https://github.com/OpenKMIP/libkmip/archive/"
+                           "refs/heads/master.zip")
+            url = self.params.get('url', default=default_url)
```

Comment:
remove default_url  as we are hard_coding it in yaml already and make default parameter as None